server/modules/category/service/categoryImportJob.js

```diff
@@ -0,0 +1,178 @@
+const R = require('ramda')
+
+const Job = require('../../../job/job')
+const CSVParser = require('../../../utils/file/csvParser')
+
+const Survey = require('../../../../common/survey/survey')
+const Category = require('../../../../common/survey/category')
+const CategoryLevel = require('../../../../common/survey/categoryLevel')
+const CategoryItem = require('../../../../common/survey/categoryItem')
+const StringUtils = require('../../../../common/stringUtils')
+
+const CategoryManager = require('../manager/categoryManager')
+const SurveyManager = require('../../survey/manager/surveyManager')
+
+const columnSuffixes = {
+  code: '_code',
+  label: '_label',
+  description: '_description',
+}
+
+const columnRegExpLabel = new RegExp(`^.*${columnSuffixes.label}(_[a-z]{2})?$`)
+const columnRegExpDescription = new RegExp(`^.*${columnSuffixes.description}(_[a-z]{2})?$`)
+
+class CategoryImportJob extends Job {
+
+  constructor (params) {
+    super(CategoryImportJob.type, params)
+
+    const { filePath } = params
+
+    this.csvParser = new CSVParser(filePath)
+
+    this.itemExtraDef = {} // category item extra props def
+    this.itemUuidByAncestorCodes = [] // cache of category items by ancestor codes
+  }
+
+  async execute () {
+    const { categoryUuid } = this.params
+
+    await this.csvParser.init()
+
+    this.total = await this.csvParser.calculateSize()
+    this.logDebug(`total size: ${this.total}`)
+
+    const surveyId = this.getSurveyId()
+
+    await CategoryManager.deleteLevelsByCategory(this.getUser(), surveyId, categoryUuid, this.tx)
+
+    this.category = await CategoryManager.fetchCategoryByUuid(surveyId, categoryUuid, true, false, this.tx)
+
+    await this._insertLevelsAndExtraPropsDef()
+
+    await this._insertItems()
```

Comment:
I would have separated the creation of the Category metadata from the insert. But this can be postponed when implementing the UI. what do you think?

Reply:
we can do it now, it makes the code more readable